packages/cli/src/commands/init/init.ts

```diff
@@ -281,6 +305,35 @@ async function createFromTemplate({
       packageName,
     });
 
+    // Update the react-native dependency if using the new @react-native-community/template.
+    // We can figure this out as it ships with react-native@1000.0.0 set to a dummy version.
+    const templatePackageJson = JSON.parse(
+      fs.readFileSync(path.join(process.cwd(), 'package.json'), 'utf8'),
+    );
+    if (templatePackageJson.dependencies?.['react-native'] === '1000.0.0') {
```

Comment:
This needs to be refined.

You would have to check also if:
- Version is set to `nightly`
- Version is set to `0.XX.0-main`

If so, do the replacing

Reply:
The template version for `react-native` is always `1000.0.0` to start with.  If it isn't then the template being worked on is one that was shipped with `react-native` (alway in versions < 0.75).

Reply:
Yup that makes sense. We then need this merged on the template then:
- https://github.com/react-native-community/template/pull/24